In [3]:
import h5py
import pandas as pds 
import numpy as np

In [46]:
def load_chx_h5(  fileanme ):
    '''May 5, 2020 Yugang devleoped for Mike at LBL to load the h5 data format
    Input:
        filename: the full data path
    Output:
        taus, 1d array, the time delay for the one-time correlation function
        g2: 2d array, one-time correlation function, shape is [number of time dalay, number of q]
        g12: 3d array, two-time correlation function, 
             shape is [number of frame, number of frame,  number of q]
        att: float sclar, the attenuation of X- 
        exp: float, exposure time of each frame (in second)
        N: the frame number in a time-series
        qr: 1d array, the values of the interested q    
    
    '''
    
    
    with h5py.File( filename, 'r') as hf:  
        #list all the available keys
        #print( list(hf.keys() ))
        #get the g2 - one time correlation function
        g2 = hf.get( 'g2'  )[:] 
        # get the tau of g2
        taus = hf.get( 'taus'  )[:] 
        #get the two time correlation function
        g12 =  hf.get( 'g12b'  )[:] 
        # get metadata
        md = {}
        M = hf.get('md')
        for key in list(M.attrs):
            md[key] = M.attrs[key]  
        att =     md['transmission']
        exp =  float( md['exposure time'] )
        N = md[ 'cam_num_images' ] 
        qr = md['qr']
        return taus, g2, g12, att, exp, N , qr
    
    
    

# An example

## giving the filanme here

In [47]:
inDir  = '/nsls2/xf11id1/analysis/2019_2/yuzhang/CHs/Results/H5/Sam11_Ca/'
fp = 'uid=358c3e41-31b0-4f4e-9271-fcf0496ca892_rings_Res.h5'


filename = inDir + fp 

## Load the h5 file

In [44]:

taus, g2, g12, att, exp, N , qr  = load_chx_h5(  fileanme )

## Check some results

In [37]:
taus

array([0.        , 0.00134   , 0.00268   , 0.00402   , 0.00536   ,
       0.0067    , 0.00804   , 0.00938   , 0.01072   , 0.0134    ,
       0.01608   , 0.01876   , 0.02144   , 0.0268    , 0.03216   ,
       0.03752   , 0.04288   , 0.0536    , 0.06432   , 0.07504   ,
       0.08576   , 0.1072    , 0.12864   , 0.15008   , 0.17151999,
       0.21439999, 0.25727999, 0.30015999, 0.34303999, 0.42879999,
       0.51455998, 0.60031998, 0.68607998, 0.85759997])

In [45]:
qr

array([0.00162, 0.0025 , 0.00339, 0.00427, 0.00516, 0.00604, 0.00692,
       0.00781, 0.00869, 0.00958, 0.01046, 0.01134, 0.01223, 0.01311,
       0.014  , 0.01488])

In [40]:
att, exp, N

(1.0, 0.00134, 1000)

In [39]:
g2.shape, g12.shape

((34, 16), (995, 995, 16))

## Some note
* One time: the taus is the time delay (the x-axis for g2 plot in the PDF report) 
* Two time: We take 1000 frames for each time sereis, but ignore the five frames in the data analysis
  this the reason that g12 gives a 995X995 shape for each q
  The x,y label for two-time will be exp*995






# Uncompress the tar.xz file

* In the linux system (two step)
    * xz -d       X.tar.xz
    * tar -xvf    X.tar

# The End!